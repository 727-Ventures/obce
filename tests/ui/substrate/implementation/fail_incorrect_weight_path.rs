mod test_pallet;

pub struct ChainExtension;

#[obce::definition]
pub trait ChainExtensionDefinition {
    fn extension_method(&mut self);
}

#[obce::implementation]
impl<'a, 'b, E, T> ChainExtensionDefinition for ExtensionContext<'a, 'b, E, T, ChainExtension>
where
    T: SysConfig + ContractConfig + crate::test_pallet::Config,
    <<T as SysConfig>::Lookup as StaticLookup>::Source: From<<T as SysConfig>::AccountId>,
    E: Ext<T = T>,
{
    #[obce(weight(dispatch = "test_method"))]
    fn extension_method(&mut self) {
        crate::test_pallet::Pallet::<T>::test_method(
            RawOrigin::Signed(self.env.ext().address().clone()).into(),
            123
        ).unwrap();
    }
}

fn main() {}
