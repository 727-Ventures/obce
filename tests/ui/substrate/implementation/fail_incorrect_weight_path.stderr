error: dispatch path should contain at least three segments
  --> tests/ui/substrate/implementation/fail_incorrect_weight_path.rs:18:19
   |
18 |     #[obce(weight(dispatch = "test_method"))]
   |                   ^^^^^^^^
